### PR TITLE
Change warning behavior in read_hdf5

### LINKED ISSUE #7725: hdf5 reader shouldn't complain for missing path for single table files
the hdf5 reader issues large amount of warnings for the missing `path` parameter even when the input file has one table. 

Maybe in that case the reader could stay silent? Also, maybe we can subclass the generic `AstropyUserWarning` to something like `AstropyUserIOWarning`, so warnings coming from other modules are not silenced off if the user gets annoyed by the IO warnings.

### PR DESCRIPTION
**Pull request by matteobachetti**
Resolves #7725

### CODE REVIEW

**Comment by bsipocz** [comment]
Maybe a bit of an overkill, but can we add a test to read in such a file with a single table and no path?

**Comment by matteobachetti** [comment]
I added one test and modified another. I added a catch_warnings call to `test_read_nopath` and verified that there is _no_ warning thrown with that specific message. Then I added `test_read_nopath_multi_tables` where I create a dataset with multiple tables and check that the warning is actually thrown.

**Comment by matteobachetti** [comment]
Ping @astrofrog

File: CHANGES.rst [hunk 1]
```diff
@@ -44,6 +44,8 @@ astropy.io.misc
 
 - Implement serialization of ``SkyCoord`` in ASDF. [#8284]
 
+- No warnings when reading HDF5 files with only one table and not ``path=`` argument [#8483]
```
Comment (astrofrog) [inline CHANGES.rst]: not -> no

**Comment by matteobachetti** [comment]
@astrofrog done

**Comment by bsipocz** [comment]
This has been already approved, so I'm merging.

Thanks @matteobachetti!

**Pull request merged**